linode/obj/helpers.go

```diff
@@ -38,23 +39,37 @@ func getObjKeysFromProvider(
 	return keys, checkObjKeysConfigured(keys)
 }
 
+func isCluster(regionOrCluster string) bool {
+	pattern := `^[a-z]{2}-[a-z]+-[0-9]+$`
+	re := regexp.MustCompile(pattern)
+	return re.MatchString(regionOrCluster)
+}
+
 // createTempKeys creates temporary Object Storage Keys to use.
 // The temporary keys are scoped only to the target cluster and bucket with limited permissions.
 // Keys only exist for the duration of the apply time.
 func createTempKeys(
 	ctx context.Context,
 	client linodego.Client,
-	bucket, cluster, permissions string,
+	bucket, regionOrCluster, permissions string,
 ) (*linodego.ObjectStorageKey, diag.Diagnostics) {
 	tflog.Debug(ctx, "Create temporary object storage access keys implicitly.")
 
+	tempBucketAccess := linodego.ObjectStorageKeyBucketAccess{
+		BucketName:  bucket,
+		Permissions: permissions,
+	}
+
+	if isCluster(regionOrCluster) {
```

Comment:
I'm a little weary about distinguishing cluster from region using regex validation, although in theory it should be fine.

I wonder if it would make sense to maybe do something along the lines of:

```go
type BucketParent struct {
	Cluster *string
	Region *string
}

func (p BucketParent) ClusterOk() (string, bool) {
	if p.Cluster != nil {
		return *p.Cluster, true
	}

	return "", false
}

func (p BucketParent) RegionOk() (string, bool) {
	if p.Region != nil {
		return *p.Region, true
	}

	return "", false
}

// ...

func createTempKeys(
	ctx context.Context,
	client linodego.Client,
	parent BucketParent,
	bucket, permissions string,
) (*linodego.ObjectStorageKey, diag.Diagnostics) {
	// ...

	if region, ok := parent.RegionOk(); ok {
		tempBucketAccess.Region = region
	}

	if cluster, ok := parent.ClusterOk(); ok {
		tempBucketAccess.Cluster = cluster
	}
}
```
type

Reply:
As discussed, this might be hard to implement due to cluster/region is read from resource ID in [some cases](https://github.com/linode/terraform-provider-linode/pull/1493/files#diff-55090792a29210600c60d24b798c34102d498a87d8b76d9e4c53098c2ad95639L60), and there isn't another obvious way to distinguish them without regex or string parsing.